File: crashpecker/src/main/java/me/peace/utils/AsyncTaskDelegate.java
package me.peace.utils;

import android.annotation.SuppressLint;
import android.os.AsyncTask;

import java.io.IOException;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;
import java.util.concurrent.RejectedExecutionException;

public class AsyncTaskDelegate {
    private static final String TAG = AsyncTaskDelegate.class.getSimpleName();
    private static ExecutorService sExecutor = Executors.newCachedThreadPool();

    @SuppressLint("NewApi")
    public static final void execute(AsyncTask<String, ?, ?> task,
                                     String... params) throws IOException {
        try {
            if (android.os.Build.VERSION.SDK_INT < android.os.Build.VERSION_CODES.HONEYCOMB) {
                task.execute(params);
            } else {
                task.executeOnExecutor(sExecutor, params);
            }
        } catch (RejectedExecutionException e) {
            LogUtils.e(TAG,"task " + task + " execute failed" + e);
            throw new IOException();
        }
    }

    @SuppressLint("NewApi")
    public static final void execute(AsyncTask<Object, ?, ?> task,
                                     Object... params) throws IOException {
        try {
            if (android.os.Build.VERSION.SDK_INT < android.os.Build.VERSION_CODES.HONEYCOMB) {
                task.execute(params);
            } else {
                task.executeOnExecutor(sExecutor, params);
            }
        } catch (RejectedExecutionException e) {
            LogUtils.e(TAG,"task " + task + " execute failed" + e);
            throw new IOException();
        }
    }
}

File: crashpecker/src/main/java/me/peace/utils/AsyncTaskDelegate.kt
package me.peace.utils

import android.annotation.SuppressLint
import android.os.AsyncTask
import android.os.Build
import java.io.IOException
import java.util.concurrent.Executors
import java.util.concurrent.RejectedExecutionException

object AsyncTaskDelegate {
    private val TAG = AsyncTaskDelegate::class.java.simpleName
    private val sExecutor = Executors.newCachedThreadPool()
    @SuppressLint("NewApi")
    @Throws(IOException::class)
    fun execute(
        task: AsyncTask<String?, *, *>,
        vararg params: String?
    ) {
        try {
            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.HONEYCOMB) {
                task.execute(*params)
            } else {
                task.executeOnExecutor(sExecutor, *params)
            }
        } catch (e: RejectedExecutionException) {
            LogUtils.e(TAG, "task $task execute failed$e")
            throw IOException()
        }
    }

    @SuppressLint("NewApi")
    @Throws(IOException::class)
    fun execute(
        task: AsyncTask<Any?, *, *>,
        vararg params: Any?
    ) {
        try {
            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.HONEYCOMB) {
                task.execute(*params)
            } else {
                task.executeOnExecutor(sExecutor, *params)
            }
        } catch (e: RejectedExecutionException) {
            LogUtils.e(TAG, "task $task execute failed$e")
            throw IOException()
        }
    }
}